hugegraph-core/src/main/java/com/baidu/hugegraph/structure/HugeElement.java

```diff
@@ -230,28 +215,45 @@ public int sizeOfProperties() {
     @Watched(prefix = "element")
     @SuppressWarnings({ "rawtypes", "unchecked" }) // (HugeProperty) propSet
     private <V> HugeProperty<V> addPropertySet(PropertyKey pkey, V value) {
-        HugeProperty<Set<V>> propSet;
+        HugeProperty<Collection<V>> propSet;
         if (this.hasProperty(pkey.id())) {
             propSet = this.getProperty(pkey.id());
         } else {
-            propSet = this.newProperty(pkey, new HashSet<V>());
+            propSet = this.newProperty(pkey, new HashSet<>());
             this.setProperty(propSet);
         }
 
+        this.addPropertyCollection(propSet, pkey, value);
+
+        // Any better ways?
+        return (HugeProperty) propSet;
+    }
+
+    @SuppressWarnings({ "rawtypes", "unchecked" })
+    private <V> void addPropertyCollection(HugeProperty<Collection<V>> property,
+                                           PropertyKey pkey, V value) {
         if (value instanceof Set) {
             E.checkArgument(pkey.checkDataType((Set) value),
                             "Invalid type of property values %s for key '%s'",
                             value, pkey.name());
-            propSet.value().addAll((Set) value);
+            property.value().addAll((Set) value);
+        } else if (value instanceof List) {
```

Comment:
merge 
two value instanceof Set and value instanceof List
to value instanceof Collection